modules/layers/src/solid-polygon-layer/solid-polygon-layer-vertex.glsl.js

```diff
@@ -45,6 +51,15 @@ varying vec4 vColor;
 varying float isValid;
 
 void main(void) {
+#ifdef IS_SIDE_VERTEX
+  vec3 positions = instancedPositions;
```

Comment:
At this point it might be easier to read to just have two vertex shader files?

Reply:
The core logic will be repeated, so any updates will have to be kept in sync across the two files. My instinct would be to avoid that, but I'll defer to your knowledge of often this code tends to be updated.

Reply:
> The core logic will be repeated, so any updates will have to be kept in sync across the two files.

A possible alternative way to deal with that is to define a third shader file, add the core logic as functions there, and import it (using string concatenation) in both layer shaders.

Done right, the extraction of the core logic to a set of reusable functions could make the code easier to read/understand/maintain.

The same technique (calling common logic functionS) can sometimes be used within a single file, isolating the ifdefs to fewer functions/places.